msgpack/src/main/scala/fs2/data/msgpack/low/model.scala

```diff
@@ -40,9 +40,18 @@ object MsgpackItem {
   case class Extension(tpe: Byte, bytes: ByteVector) extends MsgpackItem
 
   // Predefined extension types
-  case class Timestamp32(seconds: ByteVector) extends MsgpackItem
-  case class Timestamp64(nanoseconds: ByteVector, seconds: ByteVector) extends MsgpackItem
-  case class Timestamp96(nanoseconds: ByteVector, seconds: ByteVector) extends MsgpackItem
+  case class Timestamp32(seconds: Int) extends MsgpackItem
+
+  /** Stores data in a 30-bit [[nanoseconds]] and a 34-bit [[seconds]] fields, both of which are accessible as class
+    * attributes. To ensure valid data length at the type level, both fields are constructed from a single 64-bit
+    * [[combined]] variable.
+    * @param combined [[nanoseconds]] and [[seconds]] combined into a signle 64-bit value
+    */
+  case class Timestamp64(combined: Long) extends MsgpackItem {
+    val nanoseconds: Int = (combined >> 34).toInt // we are sure that (x: Long) >> 34 fits in an int
```

Comment:
When using the arithmetic right shift, it is better to apply a mask before converting.

```suggestion
    val nanoseconds: Int = (0xffffffffL & (combined >> 34)).toInt // we are sure that (x: Long) >> 34 fits in an int
```

Reply:
(note that this mask has only 7 f's and not 8 like the other one)

Reply:
I guess without the mask we could end up padding with 1s if the number is negative? Is that right?

Reply:
Yes, this is what can happen.